packages/databricks-vscode/src/configuration/ConnectionManager.ts

```diff
@@ -247,11 +246,16 @@ export class ConnectionManager implements Disposable {
 
                 await this.updateSyncDestinationMapper();
                 await this.updateClusterManager();
-                await this.configModel.set(
-                    "authProfile",
-                    authProvider.toJSON().profile as string | undefined
-                );
+                // Target may not exist when we login in the workspace without any bundle configuration
+                // (e.g. for the sake of bundle initialization wizard)
+                if (this.configModel.target) {
```

Comment:
In that case, we can avoid using this function entirely. 
1. configureLogin gives you a configured auth provider
2. We should add a authProvider.check to configureLogin. This will mean that configureLogin always gives a valid authProvider
3. Use the authProvider to initialise a new sdk client and set that as the client for a new service. 
4. Spawn a dedicated terminal for bundle init. 

This is only for the case where users do choose to make a new auth profile. When they don't we can use the existing infra from connectionManager directly. 

This also ensures any code related to bundle init remains separate from stuff that runs after init.

Reply:
My plan was to continue using the same login flow whether we are in a bundle project or not.`configureLogin` command can be called in both situations. If users then follow up with bundle-init the plan is to do logout/login sequence to pick up new bundle auth config. In most cases though bundle-init will result in us prompting to open a new window with a folder into which the project was generated.

It seems simpler if we only have one flow, but I might be missing some of your reasoning

Reply:
Improved this in the new revision